src/Compadre_Misc.hpp

```diff
@@ -51,6 +51,25 @@ int getRHSDims(DenseSolverType dense_solver_type, ConstraintType constraint_type
     }
 }
 
+KOKKOS_INLINE_FUNCTION
+void getPDims(DenseSolverType dense_solver_type, ConstraintType constraint_type, const int M, const int N, int &out_row, int &out_col) {
+    // Return the appropriate size for _P.
+    // In the case of solving with LU and additional constraint is used, _P needs
+    // to be resized to include additional row(s) based on the type of constraint.
+    if (dense_solver_type == LU) {
+        if (constraint_type == NEUMANN_GRAD_SCALAR) {
+            out_row = N + 1;
```

Comment:
M's and N's are swapped in LU case. This causes everything else to need flipped, but it should have been swapped.